webapp/app/controllers/test.js

```diff
@@ -0,0 +1,69 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  treeCompliantMetadata: function() {
+    var model = this.get('model');
+
+    var metadata = model.get('testMetadata');
+    var compliantObject = [];
+    this.createSubtree(compliantObject, '', metadata);
+    return compliantObject;
+  }.property('model.testMetadata'),
+
+  createSubtree: function myself(array_to_append, parentIndex, treeObject) {
```

Comment:
This function is a mess. I have no idea where to start if and when I'll have to maintain it. It should be heavily refactored...